source/custom/nifgen_service.custom.cpp

```diff
@@ -1,2 +1,64 @@
+#include <nifgen/nifgen_service.h>
+
+#include <stdexcept>
+
 namespace nifgen_grpc {
-}  // namespace nitclk_grpc
+
+class DriverErrorException : public std::runtime_error{
+ private:
+   int status_ = 0;
+
+ public:
+   DriverErrorException(int status) : std::runtime_error(""), status_(status) { }
+   int status() const
+   {
+     return status_;
+   }
+};
+
+static void CheckStatus(int status)
+{
+  if (status != 0) {
+    throw DriverErrorException(status);
+  }
+}
+
+//---------------------------------------------------------------------
+//---------------------------------------------------------------------
+::grpc::Status NiFgenService::CreateAdvancedArbSequence(::grpc::ServerContext* context, const CreateAdvancedArbSequenceRequest* request, CreateAdvancedArbSequenceResponse* response)
+{
+  if (context->IsCancelled()) {
+    return ::grpc::Status::CANCELLED;
+  }
+  try {
+    auto vi_grpc_session = request->vi();
+    ViSession vi = session_repository_->access_session(vi_grpc_session.id(), vi_grpc_session.name());
+    ViInt32 sequence_length = request->waveform_handles_array().size();
+    auto waveform_handles_array = const_cast<ViInt32*>(reinterpret_cast<const ViInt32*>(request->waveform_handles_array().data()));
+    auto loop_counts_array = const_cast<ViInt32*>(reinterpret_cast<const ViInt32*>(request->loop_counts_array().data()));
+    auto sample_counts_array = const_cast<ViInt32*>(reinterpret_cast<const ViInt32*>(request->sample_counts_array().data()));
+    auto marker_location_array = const_cast<ViInt32*>(reinterpret_cast<const ViInt32*>(request->marker_location_array().data()));
+
+    if (marker_location_array != NULL)
+    {
```

Comment:
Put opening bracket on the line above to keep consistent styling.

Reply:
Done